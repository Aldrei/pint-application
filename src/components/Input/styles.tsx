import { styled } from '@mui/material/styles';

export const InputStyle = styled('input')(() => ({
  marginBottom: '10px',
  height: '40px',
  minHeight: '40px',
  padding: '0 20px',
  background: 'none',
  border: '1px solid rgba(255, 255, 255, 0.23)',
  borderRadius: '4px',
  color: 'rgb(178, 186, 194)',
  '&:last-child': {
    marginBottom: '0'
  }
}));
