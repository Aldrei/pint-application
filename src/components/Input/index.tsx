import React from 'react'

import { InputStyle } from './styles'

const Input = ({...props}): React.ReactElement => {
  return <InputStyle {...props} />
}

export default Input;