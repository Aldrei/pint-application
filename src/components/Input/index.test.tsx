import { render } from '@testing-library/react';
import React from 'react';

import Input from './index';

describe('Input component', () => {
  it('Should render correctly type text with readOnly', () => {
    const nodeEl = render(<Input data-testid="test-id" name="test" type="text" value="Test do Test" readOnly />);

    const nodeElChild = nodeEl.getByTestId('test-id');

    expect(nodeElChild).toHaveAttribute('name', 'test');
    expect(nodeElChild).toHaveAttribute('type', 'text');
    expect(nodeElChild).toHaveAttribute('value', 'Test do Test');
    expect(nodeElChild).not.toBeDisabled();
    
    expect(nodeEl.baseElement).toMatchSnapshot();
  })
})