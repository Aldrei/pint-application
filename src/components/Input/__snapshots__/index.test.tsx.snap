// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should render correctly type text with readOnly 1`] = `
<body>
  <div>
    <input
      class="inputComp css-lfv6wx"
      data-testid="test-id"
      name="test"
      readonly=""
      type="text"
      value="Test do Test"
    />
  </div>
</body>
`;
